### PR TITLE
get_class wip

### PR DESCRIPTION
**Pull request by ponderingdemocritus**

### CODE REVIEW

File: crates/katana-rpc/src/lib.rs [hunk 1]
```diff
@@ -116,6 +125,24 @@ impl KatanaApiServer for KatanaRpc {
...
+        &self,
+        _block_id: BlockId,
+        _class_hash: String,
+    ) -> Result<ContractClass, Error> {
+        // get block
+
+        let state_number = self.block_number();
+
+        self.sequencer
+            .starknet_get_contract_class(&ClassHash(
```
Comment (Cheelax) [inline crates/katana-rpc/src/lib.rs:139]: we can call it get_contract_class

File: crates/katana-rpc/src/lib.rs [hunk 2]
```diff
@@ -116,6 +125,24 @@ impl KatanaApiServer for KatanaRpc {
...
     }
+
+    async fn get_class(
+        &self,
+        _block_id: BlockId,
+        _class_hash: String,
+    ) -> Result<ContractClass, Error> {
+        // get block
+
+        let state_number = self.block_number();
```
Comment (Cheelax) [inline crates/katana-rpc/src/lib.rs:136]: why do we get current block number is the expected block is passed as an argument?

File: crates/katana-rpc/src/lib.rs [hunk 3]
```diff
@@ -116,6 +125,24 @@ impl KatanaApiServer for KatanaRpc {
...
+
+        self.sequencer
+            .starknet_get_contract_class(&ClassHash(
+                StarkFelt::try_from(_class_hash.as_str()).unwrap(),
+            ))
+            .await
+            .map_err(|_| Error::from(KatanaApiError::ContractError))
+            .unwrap();
```
Comment (Cheelax) [inline crates/katana-rpc/src/lib.rs:144]: don't we need to remove the `;` here to return the value?

File: crates/katana-core/src/sequencer.rs [hunk 4]
```diff
@@ -120,6 +122,13 @@ impl KatanaSequencer {
             .unwrap()
             .get_storage_at(contract_address, storage_key)
     }
+
+    pub async fn starknet_get_contract_class(
```
Comment (tarrencev) [inline crates/katana-core/src/sequencer.rs:126]:
```suggestion
    pub async fn contract_class(
```

File: crates/katana-core/src/sequencer.rs [hunk 5]
```diff
@@ -120,6 +122,13 @@ impl KatanaSequencer {
...
+        &self,
+        class_hash: &ClassHash,
```
Comment (tarrencev) [inline crates/katana-core/src/sequencer.rs:128]:
```suggestion
        class_hash: ClassHash,
```

we can pass by value here

File: crates/katana-rpc/src/lib.rs [hunk 6]
```diff
@@ -116,6 +125,24 @@ impl KatanaApiServer for KatanaRpc {
         FieldElement::from_byte_slice_be(storage.bytes())
             .map_err(|_| Error::from(KatanaApiError::InternalServerError))
     }
+
+    async fn get_class(
+        &self,
+        _block_id: BlockId,
+        _class_hash: String,
```
Comment (tarrencev) [inline crates/katana-rpc/src/lib.rs:132]:
```suggestion
        class_hash: String,
```

Comment (ponderingdemocritus) [reply crates/katana-rpc/src/lib.rs:132]: the `get_contract_class` takes the reference not value

**Comment by ponderingdemocritus** [comment]
hitting quite a few things trying to return the seralized 'Program' - I am assuming these will be unblocked by Blockifier soon, so I am thinking of hanging it here until then. I am not sure this endpoint is critical right at this moment

**Comment by tarrencev** [comment]
yo gonna close this for now since so much has changed underneath

**Pull request closed without merging**